pywt/_multilevel.py

```diff
@@ -83,13 +86,17 @@ def wavedec(data, wavelet, mode='symmetric', level=None):
     if not isinstance(wavelet, Wavelet):
         wavelet = Wavelet(wavelet)
 
-    level = _check_level(min(data.shape), wavelet.dec_len, level)
+    try:
+        axes_shape = data.shape[axis]
```

Comment:
minor thing: with something called "shape" I tend to think it should be a tuple. In ``_check_level`` this is called ``size``, so maybe ``axis_size`` for clarity?